hermes/bin/src/runtime_extensions/hermes/cardano/event.rs

```diff
@@ -1,20 +1,29 @@
 //! Cardano Blockchain runtime extension event handler implementation.
 
 use crate::{
-    event::HermesEventPayload,
-    runtime_extensions::bindings::hermes::cardano::api::{
-        BlockSrc, CardanoBlock, CardanoBlockchainId, CardanoTxn,
+    app::ApplicationName,
+    event::{HermesEvent, HermesEventPayload, TargetApp, TargetModule},
+    runtime_extensions::bindings::{
+        exports::hermes::cardano::event_on_block::{CardanoNetwork, SubscriptionId},
+        hermes::cardano::api::Slot,
     },
+    wasm::module::ModuleId,
 };
 
 /// On Cardano block event
 pub(super) struct OnCardanoBlockEvent {
-    /// The blockchain id the block originated from.
-    pub(super) blockchain: CardanoBlockchainId,
-    /// This raw CBOR block data.
-    pub(super) block: CardanoBlock,
-    /// Source information about where the block came from, and if we are at tip or not.
-    pub(super) source: BlockSrc,
+    /// The Cardano blockchain network that the block event is originated from.
+    network: CardanoNetwork,
+    /// A unique identifier of the block subscription.
+    subscription_id: SubscriptionId,
```

Comment:
I am still thinking, maybe SubscriptionId should be a resource, with a `network` and `unsubscribe` methods attached to it?

Reply:
Will pass`subscription_id: u32`  and `block: u32` where the u32 represent the resource representation. 
Cant directly clone the resource since it is not implemented. So will construct a borrow version of these resource and pass to wasm.